### ISSUE TITLE
input mask for time

### ISSUE THREAD

**Issue by nik72619c** [sanniassin/react-input-mask#178]
Could anyone provide an example for time mask for example 12:54 AM/PM

**Comment by demkovych** [comment]
I made only for hh:mm, but you can improve it

`const TIME_MASK = [/^([0-2])/, /([0-9])/, ":", /[0-5]/, /[0-9]/]`

**Comment by oleg-prikhodko** [comment]
> I made only for hh:mm, but you can improve it
> 
> `const TIME_MASK = [/^([0-2])/, /([0-9])/, ":", /[0-5]/, /[0-9]/]`

Yes, but that way one could enter incorrect value such as **29:59**
A workaround solution would be to change second regexp based on first character of input's value
Something like this:
```javascript
function TimeInput() {
  const [time, setTime] = useState('')
  const startsWithTwo = time[0] === '2'

  const handleInput = ({ target: { value } }) => setTime(value)

  const mask = [
    /[0-2]/,
    startsWithTwo ? /[0-3]/ : /[0-9]/,
    ':',
    /[0-5]/,
    /[0-9]/
  ]
  return <InputMask mask={mask} onChange={handleInput} value={time} />
}
```

**Comment by reintroducing** [comment]
@demkovych @oleg-prikhodko How are you guys even getting regexps to work as a mask? See https://github.com/sanniassin/react-input-mask/issues/197 because I can't get it to take anything (regexp there taken directly from this issue).

**Comment by oleg-prikhodko** [comment]
> @demkovych @oleg-prikhodko How are you guys even getting regexps to work as a mask? See #197 because I can't get it to take anything (regexp there taken directly from this issue).

Seems strange. I copy-pasted regex-array from that issue but wasn't able to recreate your case. What version of React are you using it with?

**Comment by reintroducing** [comment]
@oleg-prikhodko We are using React 16.9.0 and react-input-mask 2.0.4. Is this a 3.x+ feature?

**Comment by oleg-prikhodko** [comment]
@reintroducing Looks like it. I'm on __3.0.0-alpha.0__. v2 docs says mask should be a string

**Comment by reintroducing** [comment]
ah, i think this repo master should be whats published in npm and keep 3 alpha on a branch. it makes it confusing, or there should be a note in the README that these docs are for 3.x, etc. But Im probably preaching to the choir here :)

**Comment by oleg-prikhodko** [comment]
@reintroducing Actually is is mentioned in README)
<img width="724" alt="Screenshot 2019-11-25 at 18 20 04" src="https://user-images.githubusercontent.com/25888855/69553073-906dd800-0fb0-11ea-9176-0b4f04ca8c44.png">

**Comment by reintroducing** [comment]
son of a b! i stand corrected, i just can't ready.

**Comment by comeall09** [comment]
> > I made only for hh:mm, but you can improve it
> > 
> > `const TIME_MASK = [/^([0-2])/, /([0-9])/, ":", /[0-5]/, /[0-9]/]`
> 
> Yes, but that way one could enter incorrect value such as **29:59**
> A workaround solution would be to change second regexp based on first character of input's value
> Something like this:
> ```javascript
> function TimeInput() {
>   const [time, setTime] = useState('')
>   const startsWithTwo = time[0] === '2'
> 
>   const handleInput = ({ target: { value } }) => setTime(value)
> 
>   const mask = [
>     /[0-2]/,
>     startsWithTwo ? /[0-3]/ : /[0-9]/,
>     ':',
>     /[0-5]/,
>     /[0-9]/
>   ]
>   return <InputMask mask={mask} onChange={handleInput} value={time} />
> }
> ```

I changed the second line of the mask and it works in the correct format as **23:59**
```
const mask = [
    /[0-2]/,
    /[0-3]/,
    ':',
    /[0-5]/,
    /[0-9]/
  ]
```

**Comment by kamolmachine** [comment]
> > I made only for hh:mm, but you can improve it
> > 
> > `const TIME_MASK = [/^([0-2])/, /([0-9])/, ":", /[0-5]/, /[0-9]/]`
> 
> Yes, but that way one could enter incorrect value such as **29:59**
> A workaround solution would be to change second regexp based on first character of input's value
> Something like this:
> ```javascript
> function TimeInput() {
>   const [time, setTime] = useState('')
>   const startsWithTwo = time[0] === '2'
> 
>   const handleInput = ({ target: { value } }) => setTime(value)
> 
>   const mask = [
>     /[0-2]/,
>     startsWithTwo ? /[0-3]/ : /[0-9]/,
>     ':',
>     /[0-5]/,
>     /[0-9]/
>   ]
>   return <InputMask mask={mask} onChange={handleInput} value={time} />
> }
> ```

I made some changes and now it works properly:
```javascript
const startWithTwo = value[0] === '2'
const otherCases = ['0', '1'].includes(value[0])

const mask = [
  /[0-2]/,
  startsWithTwo ? /[0-3]/ : otherCases && /[0-9]/,
  ':',
  /[0-5]/,
  /[0-9]/
]
```
